client/plots/brainImaging.js

```diff
@@ -246,154 +220,98 @@ class BrainImaging {
-				this.dataUrlF.push(result)
-			}
-			this.dom.tdF.selectAll('*').remove()
-			this.dom.imagesF = []
-			for (const [termV, result] of Object.entries(this.dataUrlF)) {
-				if (divideByTW)
-					this.dom.tdF
-						.append('div')
-						.attr('class', 'pp-chart-title')
-						.style('text-align', 'center')
-						.html('&nbsp;')
-						.style('font-weight', '600')
-						.style('font-size', '24px')
-						.style('margin-bottom', '5px')
-						.style('margin-top', '5px')
-						.style('display', 'block')
-				const img = this.dom.tdF.append('div').append('img').attr('src', result.url)
-				this.dom.imagesF.push(img)
-			}
+		this.renderLegend()
+	}
+
+	renderImages(key) {
+		const data = this.imagesData[key].data
+		const td = this.imagesData[key].td
+		this.legendValues = data.legend
+		if (data.error) throw data.error
+		const dataUrls = this.imagesData[key].dataUrls
+		for (const [termV, result] of Object.entries(data.brainImage)) {
+			dataUrls[termV] = result
 		}
-		if (this.state.updateT) {
+
+		td.selectAll('*').remove()
+		for (const [termV, result] of Object.entries(dataUrls)) {
+			if (this.state.config.divideByTW)
+				td.append('div')
+					.attr('class', 'pp-chart-title')
+					.style('text-align', 'center')
+					.text(`${termV} (n=${result.catNum})`)
+					.style('font-weight', '600')
+					.style('color', 'white')
+					.style('font-size', '24px')
+					.style('margin-bottom', '5px')
+					.style('margin-top', '5px')
+					.style('display', 'block')
+			td.append('div').append('img').attr('src', result.url)
+		}
+	}
+
+	async requestImages(keys) {
+		const promises = []
+		for (const key of keys) {
 			const body = {
 				genome: this.state.genome,
 				dslabel: this.state.dslabel,
 				refKey: this.state.config.queryKey,
-				t: this.settings.brainImageT,
+				l: this.settings[key],
 				selectedSampleFileNames: this.state.config.selectedSampleFileNames,
-				divideByTW,
-				overlayTW,
+				divideByTW: this.state.config.divideByTW,
+				overlayTW: this.state.config.overlayTW,
 				legendFilter: this.state.config.legendFilter
 			}
-			const data = await dofetch3('brainImaging', { body })
-			if (data.error) throw data.error
-
-			this.dataUrlT = []
-			for (const [termV, result] of Object.entries(data.brainImage)) {
-				this.dataUrlT.push(result)
-			}
-
-			this.dom.tdT.selectAll('*').remove()
-			this.dom.imagesT = []
-			for (const [termV, result] of Object.entries(this.dataUrlT)) {
-				if (divideByTW)
-					this.dom.tdT
-						.append('div')
-						.attr('class', 'pp-chart-title')
-						.style('text-align', 'center')
-						.html('&nbsp;')
-						.style('font-weight', '600')
-						.style('font-size', '24px')
-						.style('margin-bottom', '5px')
-						.style('margin-top', '5px')
-						.style('display', 'block')
-				const img = this.dom.tdT.append('div').append('img').attr('src', result.url)
-				this.dom.imagesT.push(img)
-			}
+			promises.push(await dofetch3('brainImaging', { body }))
```

Comment:
Do not await here when pushing to `promises` array, because that will block the loop on each iteration. Only await on the `Promise.all()`.